src/nafill.c

```diff
@@ -1,22 +1,22 @@
 #include "data.table.h"
 
-void nafillDouble(double *x, uint_fast64_t nx, unsigned int type, double fill, ans_t *ans, bool verbose) {
+void nafillDouble(double *x, uint_fast64_t nx, unsigned int type, double fill, bool nan_is_na, ans_t *ans, bool verbose) {
   double tic=0.0;
   if (verbose)
     tic = omp_get_wtime();
   if (type==0) { // const
     for (uint_fast64_t i=0; i<nx; i++) {
-      ans->dbl_v[i] = ISNA(x[i]) ? fill : x[i];
+      ans->dbl_v[i] = nan_is_na ? (ISNAN(x[i]) ? fill : x[i]) : (ISNA(x[i]) ? fill : x[i]);
```

Comment:
duplicate whole loop and use `if (nan_is_na) ... else ...` instead, to reduce number of `if` checks inside the loop

Reply:
done